homeassistant/components/uptimerobot/switch.py

```diff
@@ -0,0 +1,80 @@
+"""UptimeRobot switch platform."""
+from __future__ import annotations
+
+from typing import Any
+
+from pyuptimerobot import UptimeRobotAuthenticationException
+
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import UptimeRobotDataUpdateCoordinator
+from .const import API_ATTR_OK, DOMAIN, LOGGER
+from .entity import UptimeRobotEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the UptimeRobot switches."""
+    coordinator: UptimeRobotDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        [
+            UptimeRobotSwitch(
+                coordinator,
+                SwitchEntityDescription(
+                    key=str(monitor.id),
+                    name=f"{monitor.friendly_name} Pause",
+                    device_class=SwitchDeviceClass.SWITCH,
+                ),
+                monitor=monitor,
+            )
+            for monitor in coordinator.data
+        ],
+    )
+
+
+class UptimeRobotSwitch(UptimeRobotEntity, SwitchEntity):
+    """Representation of a UptimeRobot switch."""
+
+    _attr_icon = "mdi:cog"
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if the entity is on."""
+        return bool(self.monitor.status == 0)
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.monitor.status not in (8, 9)
+
+    async def _async_edit_monitor(self, **kwargs: Any) -> None:
+        """Edit monitor status."""
+        try:
+            response = await self.api.async_edit_monitor(**kwargs)
```

Comment:
This **will** fail for _anyone_ that has followed the documentation for this integration.
<https://www.home-assistant.io/integrations/uptimerobot>
> To get your API key, go to [My Settings](https://uptimerobot.com/dashboard#mySettings) on the UptimeRobot website, at the bottom you will find your “Read-Only API Key”.

Reply:
What approach you suggest then ? For sure update the documentation, but then ?

Reply:
Added breaking change section

Reply:
It looks like the read-only key starts with `ur` and the main key starts with just `u`, they do not say that on their documentation, but I tested with multiple accounts and recreated multiple new keys and that looks like a stable thing to check.
So we should check that during integration setup and trigger reauth if it starts with `ur`

Reply:
Check added